From a240519daa51a90feef954fe798e916c3481b2b4 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 12 Jan 2016 18:03:39 +0800
Subject: [PATCH 0841/5242] MLK-12217 video: mxsfb: fix an uninitialized
 variable usage

commit  434d3dd1a4e33b09196b231af4f00df9cf1b3aa8 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'vmode' variable is used before it is initialized.
So initialize it before that.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 601a27c..3ec3874 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1021,6 +1021,8 @@ static int mxsfb_restore_mode(struct fb_info *fb_info)
 	if (!(ctrl & CTRL_RUN))
 		return -EINVAL;
 
+	memset(&vmode, 0, sizeof(vmode));
+
 	vdctrl0 = readl(host->base + LCDC_VDCTRL0);
 	vdctrl2 = readl(host->base + LCDC_VDCTRL2);
 	vdctrl3 = readl(host->base + LCDC_VDCTRL3);
-- 
1.7.9.5

